lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -356,8 +357,11 @@ public function query()
         if ($logger) {
             $logger->startQuery($args[0]);
         }
-
-        $statement = $this->_conn->query(...$args);
+        try {
+            $statement = $this->_conn->query(...$args);
+        } catch (Throwable $ex) {
+            throw DBALException::driverExceptionDuringQuery($this->_driver, $ex, $args[0]);
```

Comment:
Although this is correct, this is a breaking change and should be retargeted against `3.0.x`.